simpleviewinc/keg-hub
# Consumer Container Updates

> Note: this PR is paired with [this PR](https://github.com/simpleviewinc/tap-events-force/pull/50) from `tap-events-force`. They have the same testing instructions, and either can be reviewed/merged first

# Context
* we need to be frequently testing the tap with `keg-test-consumer` to verify the build works as expected
* right now that's a laborious process

# Goal
* speed up testing 
* run everything in docker
* add keg-test-consumer to the hub as another repo

# Updates
* migrated `keg-test-consumer` into `keg-hub/repos`
* added a `README.md` that provides testing instructions
* added container folder with injected app files
* updated `package.json` with some scripts for testing taps

# Testing
* `keg evf && keg pr 50`
* `keg hub && keg pr 25`

1. `keg consumer start`
  * starts the container itself, but not the react app yet, with the consumer mounted at `keg/app`
2. `keg consumer sync evf:install:start`
  * syncs your local tap-events-force into the container at `/keg/tap`, then starts the build server, which will rebuild the sessions component whenever a change is detected. It will automatically
    place it inside of the consumer's `/keg/app/node_modules/@keg-hub/tap-evf-sessions` 
3. `keg consumer att`
  * attach to the consumer container
4. `yarn start:dev`
  * starts the react app, but also restarts the app if any changes to the sessions component build are detected
5. Verify you can see the sessions component at `local.kegdev.xyz`
6. Make a change in the tap
7. Verify you see it rebuilding the sessions component in the terminal output from step 2
8. Verify you see the consumer terminal output from step 4 restarting the server
9. Refresh the `local.kegdev.xyz` browser tab, and verify you see your change


> Note: I'd prefer if we could just use hot reloading, rather than restarting the whole consumer server. The issue I encountered was I could not initiate it. I tried adding a file to the consumer's directory, but because of the mutagen one-way-sync, that was rejected and the file never appears in the directory. We also need it to use the latest build, which I'm not sure hot reloading would do. I'm open to other solutions here if you have any.
